R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
```

Comment:
Is there way to check if an env variable is set in R? I don't expect it to come up in this case, but with this pattern it seems like it would be impossible for a user to set `DASH_REQUESTS_PATHNAME_PREFIX=""`, as we interpret `""` the same as the user didn't set it.
Shouldn't affect this case, but as we use this same type of config variable resolution for other variables, it might come up

Reply:
Yeah, the current approach is no good from that point of view. Yes, here's one possible approach:

```
> "DASH_REQUESTS_PATHNAME_PREFIX" %in% names(Sys.getenv())
[1] FALSE
> "HOME" %in% names(Sys.getenv())
[1] TRUE
> Sys.getenv("HOME")
[1] "/Users/rpkyle"
```

This may not work reliably across all platforms, we'd probably want to test:

>_On most platforms_ `Sys.getenv()` _will return a named vector giving the values of all the environment variables, sorted in the current locale. It may be confused by names containing = which some platforms allow but POSIX does not. (Windows is such a platform: there names including = are truncated just before the first =.)_